app/models/katello/repository.rb

```diff
@@ -902,6 +903,17 @@ def ensure_valid_ignorable_content
       end
     end
 
+    def ensure_valid_upstream_authorization
+      return if (self.upstream_username.blank? && self.upstream_password.blank?)
+      if redhat?
+        errors.add(:base, N_("Upstream username and password may only be set on custom repositories."))
+      elsif !self.upstream_password
```

Comment:
should this be  elsif self.upstream_password.blank?

(same for below)

Reply:
actually, should we allow a blank password? that seems valid.  In which case below should be self.upstream_username.blank?   and this check should be removed.

Reply:
A blank password is allowed by Pulp, but not passing in any password is not allowed.

For example:

```
$ hammer -u admin -p changeme --verify-ssl false repository update --id 5 --upstream-username baz
Could not update the repository:
  Task 6ef92ff6-9375-413a-be1f-de345896fa6e: Katello::Errors::PulpError: PLP0000: Configuration errors:
  The configuration parameter <basic_auth_username> requires the <basic_auth_password> parameter to also be set.
```
I'm going to allow Katello to submit a username without pass anyway since this PR allows Pulp's error to be raised back to the user.

Reply:
I ended up keeping it the way it was before, it's better for Katello to error off the bat vs sending through to Pulp and having a task fail.